lib/python/qmk/keyboard.py

```diff
@@ -216,3 +199,64 @@ def render_layouts(info_json, render_ascii):
         layouts[layout] = render_layout(layout_data, render_ascii)
 
     return layouts
+
+
+def render_key_rect(textpad, x, y, w, h, label, style):
+    box_chars = BOX_DRAWING_CHARACTERS[style]
+    x = ceil(x * 4)
+    y = ceil(y * 3)
+    w = ceil(w * 4)
+    h = ceil(h * 3)
+
+    label_len = w - 2
+    label_leftover = label_len - len(label)
+
+    if len(label) > label_len:
+        label = label[:label_len]
+
+    label_blank = ' ' * label_len
+    label_border = box_chars['h'] * label_len
+    label_middle = label + ' '*label_leftover  # noqa: yapf insists there be no whitespace around *
+
+    top_line = array('u', box_chars['tl'] + label_border + box_chars['tr'])
+    lab_line = array('u', box_chars['v'] + label_middle + box_chars['v'])
+    mid_line = array('u', box_chars['v'] + label_blank + box_chars['v'])
+    bot_line = array('u', box_chars['bl'] + label_border + box_chars['br'])
+
+    textpad[y][x:x + w] = top_line
+    textpad[y + 1][x:x + w] = lab_line
+    for i in range(h - 3):
+        textpad[y + i + 2][x:x + w] = mid_line
+    textpad[y + h - 1][x:x + w] = bot_line
+
+
+def render_key_isoenter(textpad, x, y, w, h, label, style):
+    box_chars = BOX_DRAWING_CHARACTERS[style]
+    x = ceil(x * 4)
+    y = ceil(y * 3)
+    w = ceil(w * 4)
+    h = ceil(h * 3)
+
+    label_len = w - 1
+    label_leftover = label_len - len(label)
+
+    if len(label) > label_len:
+        label = label[:label_len]
+
+    label_blank = ' ' * (label_len-1)  # noqa: yapf insists there be no whitespace around - and *
+    label_border_top = box_chars['h'] * label_len
+    label_border_bottom = box_chars['h'] * (label_len-1)  # noqa
+    label_middle = label + ' '*label_leftover  # noqa
+
+    top_line = array('u', box_chars['tl'] + label_border_top + box_chars['tr'])
+    lab_line = array('u', box_chars['v'] + label_middle + box_chars['v'])
+    crn_line = array('u', box_chars['bl'] + box_chars['tr'] + label_blank + box_chars['v'])
+    mid_line = array('u', box_chars['v'] + label_blank + box_chars['v'])
+    bot_line = array('u', box_chars['bl'] + label_border_bottom + box_chars['br'])
+
+    textpad[y][x - 1:x + w + 1] = top_line
+    textpad[y + 1][x - 1:x + w + 1] = lab_line
+    textpad[y + 2][x - 1:x + w + 1] = crn_line
```

Comment:
```suggestion
    textpad[y][x - 1:x + w] = top_line
    textpad[y + 1][x - 1:x + w] = lab_line
    textpad[y + 2][x - 1:x + w] = crn_line
```

Reply:
Nope, I think that's just left over from when I was trying to get it working and experimenting with offsets. All the lines should end at the same column.